### PR TITLE
[Merged by Bors] - feat: port Algebra.Order.Pointwise

### PR DESCRIPTION
**Pull request by qawbecrdtey**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Order/Pointwise.lean [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+variable [One α]
+
+-- Porting note: Use something like `@[to_additive (attr := simp)]` instead.
+-- However, this makes a simpNF error like:
+--  `csupₛ_one` can be proved like `by simp only [csupₛ_zero]
+-- So the `simp` attribute is written below the definition.
+@[to_additive]
+theorem csupₛ_one : supₛ (1 : Set α) = 1 :=
+  csupₛ_singleton _
+#align csupₛ_one csupₛ_one
```
Comment (jcommelin) [inline Mathlib/Algebra/Order/Pointwise.lean:49]: Please add additivized versions of `#align` everywhere.

File: Mathlib/Algebra/Order/Pointwise.lean [hunk 2]
```diff
@@ -106,15 +121,30 @@ section One
...
 variable [One α]
 
-@[simp, to_additive]
-theorem Sup_one : supₛ (1 : Set α) = 1 :=
+-- Porting note: Use something like `@[to_additive (attr := simp)]` instead.
+-- However, this makes a simpNF error like:
+--  `supₛ_one` can be proved like `by simp only [@supₛ_zero]`
+-- So the `simp` attribute is written below the definition.
+@[to_additive]
+theorem supₛ_one : supₛ (1 : Set α) = 1 :=
```
Comment (jcommelin) [inline Mathlib/Algebra/Order/Pointwise.lean:129]: I think we should try to solve this in a different way. I asked on zulip: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/simpNF/near/321179015

File: Mathlib/Algebra/Order/Pointwise.lean [hunk 3]
```diff
@@ -106,15 +121,30 @@ section One
...
+#align Inf_zero infₛ_zero
+#align Inf_one infₛ_one
+
+attribute [simp] supₛ_zero
+attribute [simp] infₛ_zero
+attribute [simp] supₛ_one
+attribute [simp] infₛ_one
+attribute [simp] csupₛ_zero
+attribute [simp] cinfₛ_zero
+attribute [simp 900] csupₛ_one
```
Comment (jcommelin) [inline Mathlib/Algebra/Order/Pointwise.lean:146]: Why these priorities?

File: Mathlib/Algebra/Order/Pointwise.lean (hunk 2 above)
Comment (eric-wieser) [reply Mathlib/Algebra/Order/Pointwise.lean:129]: I can't reproduce this `simpNF` error.

Comment (qawbecrdtey) [reply Mathlib/Algebra/Order/Pointwise.lean:129]: @eric-wieser The attributes below should be removed to reproduce the error, with the following code:

Comment (qawbecrdtey) [reply Mathlib/Algebra/Order/Pointwise.lean]: It seems like swapping the place of two sections `CompleteLattice` and `ConditionallyCompleteLattice` fixes the issue. Thanks for reviewing, I could fix it.

File: Mathlib/Algebra/Order/Pointwise.lean [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+
+/-!
+# Pointwise operations on ordered algebraic objects
+
+This file contains lemmas about the effect of pointwise operations on sets with an order structure.
+
+## TODO
+
+`Sup (s • t) = Sup s • Sup t` and `Inf (s • t) = Inf s • Inf t` hold as well but
+`covariant_class` is currently not polymorphic enough to state it.
```
Comment (jcommelin) [inline Mathlib/Algebra/Order/Pointwise.lean:24]:
```suggestion
`CovariantClass` is currently not polymorphic enough to state it.
```

File: Mathlib/Algebra/Order/Pointwise.lean [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+    exact (mul_le_mul_left hr).mpr a_h_left
+  · rintro h
+    use x / r
+    constructor
+    exact (div_le_iff' hr).mpr h
+    exact mul_div_cancel' _ (ne_of_gt hr)
+#align linear_ordered_field.smul_Iic LinearOrderedField.smul_Iic
+
+end LinearOrderedField
+#lint
```
Comment (jcommelin) [inline Mathlib/Algebra/Order/Pointwise.lean:297]:
```suggestion

```

File: Mathlib/Algebra/Order/Pointwise.lean [hunk 6]
```diff
@@ -9,7 +9,9 @@ Authors: Alex J. Best, Yaël Dillies
 ! if you have ported upstream changes.
 -/
 import Mathlib.Algebra.Bounds
-import Mathlib.Data.Set.Pointwise.Smul
+import Mathlib.Algebra.Order.Field.Basic
```
Comment (jcommelin) [inline Mathlib/Algebra/Order/Pointwise.lean:12]: Why do we need the new imports?

Comment (jcommelin) [reply Mathlib/Algebra/Order/Pointwise.lean:12]: Why do we need the new import?

File: Mathlib/Algebra/Order/Pointwise.lean [hunk 7]
```diff
@@ -29,148 +30,161 @@ open Pointwise
...
+-- due to simpNF problem between `supₛ_xx` `csupₛ_xx`.
 
-variable [ConditionallyCompleteLattice α]
+section CompleteLattice
+
+variable [CompleteLattice α]
 
 section One
 
 variable [One α]
```
Comment (jcommelin) [inline Mathlib/Algebra/Order/Pointwise.lean:42]:
```suggestion
variable [One α]

```

**Comment by jcommelin** [comment]
There is still a new import... if we need it, please explain why.

File: Mathlib/Algebra/Order/Pointwise.lean (hunk 6 above)
Comment (qawbecrdtey) [reply Mathlib/Algebra/Order/Pointwise.lean:12]: If you mean `Mathlib.Data.Set.Pointwise.SMul`, that is required for, for example, `mem_smul_set`.

Comment (qawbecrdtey) [reply Mathlib/Algebra/Order/Pointwise.lean:12]: And `Mathlib.Algebra.Order.Field.Basic` is for instance `LinearOrderedField`.

Comment (qawbecrdtey) [reply Mathlib/Algebra/Order/Pointwise.lean]: Thanks for all those help, I'll be more careful afterwards.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**